exchange/order/limits/levels.go

```diff
@@ -0,0 +1,180 @@
+		if !dAmount.Mod(dStep).IsZero() {
+			return fmt.Errorf("%w stepSize: %.8f supplied %.8f",
+				ErrAmountExceedsStep,
+				m.AmountStepIncrementSize,
+				amount)
+		}
+	}
+
+	// ContractMultiplier checking not done due to the fact we need coherence with the
+	// last average price (TODO)
+	// m.multiplierUp will be used to determine how far our price can go up
+	// m.multiplierDown will be used to determine how far our price can go down
+	// m.averagePriceMinutes will be used to determine mean over this period
+
+	// Max iceberg parts checking not done as we do not have that
+	// functionality yet (TODO)
+	// m.maxIcebergParts // How many components in an iceberg order
+
+	// Max total orders not done due to order manager limitations (TODO)
+	// m.maxTotalOrders
+
+	// Max algo orders not done due to order manager limitations (TODO)
+	// m.maxAlgoOrders
+
+	// If order type is Market we do not need to do price checks
+	if orderType != order.Market {
+		if m.MinPrice != 0 && price < m.MinPrice {
+			return fmt.Errorf("%w min: %.8f supplied %.8f",
+				ErrPriceBelowMin,
+				m.MinPrice,
+				price)
+		}
+		if m.MaxPrice != 0 && price > m.MaxPrice {
+			return fmt.Errorf("%w max: %.8f supplied %.8f",
+				ErrPriceExceedsMax,
+				m.MaxPrice,
+				price)
+		}
+		if m.MinNotional != 0 && (amount*price) < m.MinNotional {
+			return fmt.Errorf("%w minimum notional: %.8f value of order %.8f",
+				ErrNotionalValue,
+				m.MinNotional,
+				amount*price)
+		}
+		if m.PriceStepIncrementSize != 0 {
+			dPrice := decimal.NewFromFloat(price)
+			dMinPrice := decimal.NewFromFloat(m.MinPrice)
+			dStep := decimal.NewFromFloat(m.PriceStepIncrementSize)
+			if !dPrice.Sub(dMinPrice).Mod(dStep).IsZero() {
+				return fmt.Errorf("%w stepSize: %.8f supplied %.8f",
+					ErrPriceExceedsStep,
+					m.PriceStepIncrementSize,
+					price)
+			}
+		}
+		return nil
+	}
+
+	if m.MarketMinQty != 0 &&
+		m.MinimumBaseAmount < m.MarketMinQty &&
+		amount < m.MarketMinQty {
+		return fmt.Errorf("%w min: %.8f supplied %.8f",
+			ErrMarketAmountBelowMin,
+			m.MarketMinQty,
+			amount)
+	}
+	if m.MarketMaxQty != 0 &&
+		m.MaximumBaseAmount > m.MarketMaxQty &&
+		amount > m.MarketMaxQty {
+		return fmt.Errorf("%w max: %.8f supplied %.8f",
+			ErrMarketAmountExceedsMax,
+			m.MarketMaxQty,
+			amount)
+	}
+	if m.MarketStepIncrementSize != 0 &&
+		m.AmountStepIncrementSize != m.MarketStepIncrementSize {
+		dAmount := decimal.NewFromFloat(amount)
+		dMinMAmount := decimal.NewFromFloat(m.MarketMinQty)
+		dStep := decimal.NewFromFloat(m.MarketStepIncrementSize)
+		if !dAmount.Sub(dMinMAmount).Mod(dStep).IsZero() {
+			return fmt.Errorf("%w stepSize: %.8f supplied %.8f",
+				ErrMarketAmountExceedsStep,
+				m.MarketStepIncrementSize,
+				amount)
+		}
+	}
+	return nil
+}
+
+// ConformToDecimalAmount (POC) conforms amount to its amount interval
+func (m *MinMaxLevel) ConformToDecimalAmount(amount decimal.Decimal) decimal.Decimal {
```

Comment:
🚧   I think Conform is actually `Floor` ( as opposed to `Round` )
🦍 It sucks, but I don't think `ToDecimalAmount` is verbose enough
🤷‍♂️ Is POC .. Proof of Concept?
🔁 Also applies to others below.
```suggestion
// FloorAmountToDecimalStepInterval rounds the amount to the AmountStepIncrementSize
func (m *MinMaxLevel) FloorToDecimalStepInterval(amount decimal.Decimal) decimal.Decimal {
```